### PR TITLE
Assignment 4 - Authentication

### LINKED ISSUE #118: Basic authentication
Using the `hapi-auth-basic` plugin, add a new `/private` endpoint protected by a username and password that returns a simple HTML page with a welcome message. The welcome message should include the logged in username. Do not use any templates (views) at this stage. For username/password validation, create a static `lib/users.json` file (we will replace this in the near future) that contains a few sample users and passwords.

The new endpoint will be contained in its own plugin, similar to how we set up `/version`, and will require bringing test coverage back to 100%.

Due: 4/27

### PR DESCRIPTION
**Pull request by tielur**
closes #118

### CODE REVIEW

File: lib/private.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  return callback(null, isValid, { username: user.username});
+};
+
+
+exports.register = function (server, options, next) {
+
+  server.register(HapiAuthBasic, function (err) {
+
+    if (err) {
+      return next(err);
```
Comment (hussainanjar) [inline lib/private.js]: Your indentation is not 4 spaces consistently.

Comment (tielur) [inline lib/private.js]: Thanks @hueniverse. I ended up copying the .eslintrc and .eslintignore from lab and putting it in the root directory of this project. I then turned on indentationn to match the style we are trying to follow:

```
"indent": [2, 4]
```

http://eslint.org/docs/rules/indent.html


Is it common to do this with Lab? Lab has a default for both of these files, but to add a new rule or change rules, you just copy it and make modifications? Should these be committed to the project?

**Comment by tielur** [comment]
I opened this pull request over at lab to hopefully catch things like this in the future.

https://github.com/hapijs/lab/pull/349

**Comment by hussainanjar** [comment]
@tielur :+1:

**Comment by tielur** [comment]
Lab was updated, so you should now get a warning for indentation.  Something like this:

```
Linting results:
	lib/index.js:
		Line 15: indent - Expected indentation of 4 characters.
```

**Comment by FennNaten** [comment]
Nice!

File: lib/private.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+internals.validate = function (username, password, callback) {
+
+    var user = Users[username];
+    if (!user) {
+        return callback(null, false);
+    }
+
+    var isValid = password === user.password;
+
+    return callback(null, isValid, { username: user.username});
```
Comment (idanwe) [inline lib/private.js]: missing before `}`

**Pull request closed without merging**